objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+                local xOffset = -9.5
+                local xScale = 19
+                local xSpacing = 1/math.max(5, #toPlayFast + #toPlaySlow - 1)
+                local yOffset = -0.22
+                local ySpacing = 0.05
+                local zOffset = 16
+                for ind,card in pairs(toPlayFast) do
+                    card.use_hands = false
+                    card.setPositionSmooth(spos + Vector(xOffset+xScale*(ind-1)*xSpacing, yOffset+(ind-1)*ySpacing, zOffset), false, false)
+                    card.setRotation({0,180,0})
+                    Wait.condition(
+                        function() card.use_hands = true end,
+                        function() return not card.isSmoothMoving() end,
+                        3, -- don't think this timeout function will be necessary, but thought I'd add it to be safe
+                        function()
+                            card.use_hands = true
+                            card.setPosition(spos + Vector(xOffset+xScale*(ind-1)*xSpacing, yOffset+(ind-1)*ySpacing, zOffset))
+                        end
+                    )
+                end
+                for ind,card in pairs(toPlaySlow) do
+                    card.use_hands = false
+                    card.setPositionSmooth(spos + Vector(xOffset+xScale*(ind+#toPlayFast-1)*xSpacing, yOffset+(ind+#toPlayFast-1)*ySpacing, zOffset), false, false)
+                    card.setRotation({0,180,0})
+                    Wait.condition(
+                        function() card.use_hands = true end,
+                        function() return not card.isSmoothMoving() end,
+                        3, -- don't think this timeout function will be necessary, but thought I'd add it to be safe
+                        function()
+                            card.use_hands = true
+                            card.setPosition(spos + Vector(xOffset+xScale*(ind-1)*xSpacing, yOffset+(ind-1)*ySpacing, zOffset))
+                        end
+                    )
+                end
+            end
+        end
+    end
+end
+
+function onObjectEnterContainer(cont, obj)
+    local guid = obj.guid
+    if impendTable[guid] then
+        impendTable[guid].container = cont.guid
+    end
+    updateSave()
+end
+
+function onObjectLeaveContainer(_container, obj)
+    if impendTable[obj.guid] then
+        impendTable[obj.guid].container = nil
+        Wait.time(function() -- needs a wait otherwise the card's energy variable is nil
+            createCardButtons(obj)
+            updateEnergyDisplay(obj.guid)
+            if g3Selected and impendTable[obj.guid].turnSelected ~= Global.getVar("turn") then
+                createG3Buttons(obj)
+                updateG3Buttons(obj.guid)
+            end
+        end, 0.01)
+    end
+    updateSave()
+end
+
+function modifyCost(params)
+    local costs = params.costs
+    if params.color == Global.call("getSpiritColor", {name = spiritName}) and costs ~= nil then
```

Comment:
was `costs` nil for you at some point? or just preemptive guarding?

Reply:
When `modifyCost()` was first introduced I encountered an error or two that I reported to pyrrhic and this was presumably a workaround that is now just likely a relic that can be removed.